### ISSUE TITLE
Nearest neighbors result mismatch

### ISSUE THREAD

**Issue by jverce** [Stoeoef/spade#54]
As per the docs, my assumption is that `RTree::nearest_neighbors` should behave exactly like `RTree:nearest_neighbor` whenever there is a single nearest neighbor for a given point.

However I found a mismatch that's breaking my program:
```rust
        // `prev` is some arbitrary point that used
        // to belong to `rtree` but now it doesn't.
        let mut nns = rtree.nearest_neighbors(&prev);
        nns.sort();
        let nn1 = nns.iter().nth(0).unwrap();

        let nn = rtree.nearest_neighbor(&prev).unwrap();

        println!("query_point={:?}", prev);
        println!("nns={:?}", nns);
        println!("nn={:?}", nn);

        assert_eq!(*nn1, nn);
```

The assertion usually succeeds (which is expected), but sometimes it doesn't. Here's an example output of when it fails:

```
query_point=PointVertex { point: [486.5296335632694, 3987.037159235611], id: 1545 }
nns=[PointVertex { point: [484.4154017435969, 3914.9206580969], id: 1563 }]
nn=PointVertex { point: [449.87221229993327, 3986.728686301482], id: 2981 }
thread 'solution_is_computed_correctly_tsp_heuristic' panicked at 'assertion failed: `(left == right)`
  left: `PointVertex { point: [484.4154017435969, 3914.9206580969], id: 1563 }`,
 right: `PointVertex { point: [449.87221229993327, 3986.728686301482], id: 2981 }`', src/week3/solution.rs:24:9
```

As mentioned before, the assertion usually succeeds, e.g.:
```
query_point=PointVertex { point: [555.0393744191871, 3883.5123282564214], id: 567 }
nns=[PointVertex { point: [537.4013213059632, 3945.0350382362126], id: 3543 }]
nn=PointVertex { point: [537.4013213059632, 3945.0350382362126], id: 3543 }
query_point=PointVertex { point: [537.4013213059632, 3945.0350382362126], id: 3543 }
nns=[PointVertex { point: [518.5897669888586, 3953.4962614450205], id: 3200 }]
nn=PointVertex { point: [518.5897669888586, 3953.4962614450205], id: 3200 }
query_point=PointVertex { point: [518.5897669888586, 3953.4962614450205], id: 3200 }
nns=[PointVertex { point: [486.5296335632694, 3987.037159235611], id: 1545 }]
nn=PointVertex { point: [486.5296335632694, 3987.037159235611], id: 1545 }
```

Is this a misunderstanding on my part? Or is there a bug to be fixed?

**Comment by Stoeoef** [comment]
Thanks a lot for finding, reporting and fixing this bug!

Version 1.8.2 has just been published, containing the bugfix.

Just as an FYI: In case you're running something performance critical code, you may want to switch to [rstar](https://github.com/Stoeoef/rstar) which is faster. It's not a drop in replacement, though.

I also checked: The bug you reported has not been ported over!

**Comment by Stoeoef** [comment]
Closed with #55

**Comment by jverce** [comment]
@Stoeoef re: **rstar** crate, yeah, I saw it this morning. I did need the `nearest_neighbors` API though, so I wouldn't have been able to use it even if I knew it in advance. I'll try to find some time to implement it in that other crate as well (created an issue for that https://github.com/Stoeoef/rstar/issues/30).
